### PR TITLE
feat: added support for enabling copying by foldername

### LINKED ISSUE #15338: "minikube cp" dosnt work if you dont give full file name in destination
### What Happened?

if you do 
```
$ minikube cp ~/Desktop/episode1/Final.mov /tmp/
```
it wont work

but if you do 
```
$ minikube cp ~/Desktop/episode1/Final.mov /tmp/Final.mov 
```
it works

we should either support it or let the user they need to specify the full destination file name

more logs 

```
$ minikube cp ~/Desktop/episode1/Final.mov /tmp/ --alsologtostderr
I1109 16:06:40.147264   39834 out.go:296] Setting OutFile to fd 1 ...
I1109 16:06:40.147539   39834 out.go:348] isatty.IsTerminal(1) = true
I1109 16:06:40.147562   39834 out.go:309] Setting ErrFile to fd 2...
I1109 16:06:40.147572   39834 out.go:348] isatty.IsTerminal(2) = true
I1109 16:06:40.147796   39834 root.go:334] Updating PATH: /Users/medya/.minikube/bin
I1109 16:06:40.148407   39834 mustload.go:65] Loading cluster: minikube
I1109 16:06:40.148626   39834 config.go:180] Loaded profile config "minikube": Driver=qemu2, ContainerRuntime=docker, KubernetesVersion=v1.25.3
I1109 16:06:40.149482   39834 host.go:66] Checking if "minikube" exists ...
I1109 16:06:40.149710   39834 ssh_runner.go:195] Run: stat -c "%s %y" /tmp/tmp
I1109 16:06:40.149722   39834 sshutil.go:53] new ssh client: &{IP:192.168.105.11 Port:22 SSHKeyPath:/Users/medya/.minikube/machines/minikube/id_rsa Username:docker}
I1109 16:06:40.176717   39834 ssh_runner.go:356] copy: skipping /tmp/tmp (exists)

```

### PR DESCRIPTION
**Pull request by NimishKashyap**
fixes #15338

Changes - 
- Added support for copying to destination without specifying filename.

```
minikube cp path/to/file.txt desination/folder/
```

### CODE REVIEW

**Comment by NimishKashyap** [comment]
> @NimishKashyap could you please add Before and After this PR ? and also how about adding an integration test in our functional test for this ?
> 
> here is link to our CP test, you could add more test to ensure this feature doesnt break https://github.com/medyagh/minikube/blob/1740a4de3582b86da41df024641cdab7cb5766b7/test/integration/functional_test.go#L1687

Alright I'll take a look into it

**Comment by NimishKashyap** [comment]
> @NimishKashyap could you please add Before and After this PR ? and also how about adding an integration test in our functional test for this ?
> 
> here is link to our CP test, you could add more test to ensure this feature doesnt break https://github.com/medyagh/minikube/blob/1740a4de3582b86da41df024641cdab7cb5766b7/test/integration/functional_test.go#L1687

I have added basic test for the command.
Before: 
```
minikube cp /folder/filename.txt /destFolder/filename.txt
```
After:
```
minikube cp /folder/filename.txt /destFolder/
```

This is my first time contributing to open source.

File: cmd/minikube/cmd/cp.go [hunk 1]
```diff
@@ -56,8 +56,27 @@ Example Command : "minikube cp a.txt /home/docker/b.txt" +
...
-		src := newRemotePath(args[0])
-		dst := newRemotePath(args[1])
+		// This check has been introduced to enable support to copy using the following syntax
+		// minikube cp <HOST>/tmp/filename.xyz /tmp/
+
+		sourcePathList := strings.Split(args[0], "/")
+		destinationPathList := strings.Split(args[1], "/")
+
+		sourceFileName := sourcePathList[len(sourcePathList)-1]
+		destinationFileName := destinationPathList[len(destinationPathList)-1]
```
Comment (Aut0R3V) [inline cmd/minikube/cmd/cp.go:66]: Looks okay to me

File: test/integration/licenses/cloud.google.com/go/compute/metadata/LICENSE [hunk 2]
```diff
@@ -0,0 +1,202 @@
+
```
Comment (medyagh) [inline test/integration/licenses/cloud.google.com/go/compute/metadata/LICENSE:1]: @NimishKashyap there are many unrelated files that were checked in please remove them and also ensure it is not in the commit history

Comment (NimishKashyap) [reply test/integration/licenses/cloud.google.com/go/compute/metadata/LICENSE]: I have deleted those files. Please take a look into it.

File: .gitignore [hunk 3]
```diff
@@ -8,6 +8,8 @@
 _obj
 _test
 
+test/integration/licenses/
```
Comment (spowelljr) [inline .gitignore:11]: I'm fixing the `licenses` dir not being removed with this PR here: https://github.com/kubernetes/minikube/pull/15376

So could you remove this line and instead just run `rm -rf test/integration/licenses` assuming your on Unix

**Review by spowelljr — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 3 above)
Comment (NimishKashyap) [reply .gitignore:11]: Okay will do!

**Comment by NimishKashyap** [comment]
I'm facing issues with adding the tests. I believe my implementation should work but I'm not sure about adding the tests for that. Can anyone guide me or point out some resources to add the tests correctly? @medyagh
